lib/usd/ui/USDImportDialogCmd.cpp

```diff
@@ -93,30 +166,55 @@ MStatus USDImportDialogCmd::doIt(const MArgList& args)
 		return MS::kSuccess;
 	}
 
+	if(argData.isFlagSet(kApplyToProxyFlag))
+	{
+		MStringArray proxyArray;
+		st = argData.getObjects(proxyArray);
+		if (!st || proxyArray.length() != 1)
+			return MS::kInvalidParameter;
+
+		return applyToProxy(proxyArray[0]);
+	}
+
 	MStringArray filenameArray;
 	st = argData.getObjects(filenameArray);
 	if (st && (filenameArray.length() > 0))
 	{
 		// We only use the first one.
 		MFileObject fo;
+		MString assetPath;
 		fo.setRawFullName(filenameArray[0]);
-		if (fo.exists())
+		bool validTarget = fo.exists();
+		if (!validTarget) {
+			// Give the default usd-asset-resolver a chance
+			if (const char* cStr = filenameArray[0].asChar()) {
+				validTarget = !ArGetResolver().Resolve(cStr).empty();
+				if (validTarget)
+					assetPath = filenameArray[0];
+			}
+		} else
+			assetPath = fo.resolvedFullName();
+
+		if (validTarget)
 		{
 			USDQtUtil usdQtUtil;
 			ImportData& importData = ImportData::instance();
-			MString usdFile = fo.resolvedFullName();
-			std::unique_ptr<IUSDImportView> usdImportDialog(new USDImportDialog(usdFile.asChar(), &importData, usdQtUtil, MQtUtil::mainWindow()));
-			if (usdImportDialog->execute())
-			{
-				// The user clicked 'Apply' so copy the info from the dialog to the import data instance.
-				importData.setFilename(usdImportDialog->filename());
-				importData.setStageInitialLoadSet(usdImportDialog->stageInitialLoadSet());
-				importData.setRootPrimPath(usdImportDialog->rootPrimPath());
-				// Don't set the stage pop mask until we solve how to use it together with
-				// the root prim path.
-				//importData.setStagePopulationMask(usdImportDialog->stagePopulationMask());
-				importData.setPrimVariantSelections(usdImportDialog->primVariantSelections());
-			}
+			std::unique_ptr<IUSDImportView> usdImportDialog(
+				new USDImportDialog(assetPath.asChar(), &importData,
+					                usdQtUtil, MQtUtil::mainWindow()));
+			if (!usdImportDialog->execute())
+				return MS::kFailure;
```

Comment:
According to the doc for the execute() interface method: "True if the user applied the changes, false if they canceled." We do not want to return an error from the command when they cancel. Thus the previous if check (to set the importData) and then always return MS::kSuccess.